index.bs

```diff
@@ -386,12 +388,61 @@ To <dfn abstract-op>initialize a global object's embedder policy from a response
                 item's [=embedder policy/report only reporting endpoint=] is non-null, then
                 |policy|'s [=embedder policy/report only reporting endpoint=] to the item's
                 [=embedder policy/report only reporting endpoint=].
+4.  Otherwise:
+
+    1.   Set |policy| to |response policy|.
+
+5.  Set |global|'s [=WorkerGlobalScope/embedder policy=] to |policy|.
+
+### Checking a global object's Embedder policy ### {#check-embedder-policy-for-global}
+<div algorithm="to check a global object's embedder policy">
+To <dfn abstract-op lt="queue coep worker violation">Queue a Cross-Origin Embedder Policy
+violation on worker initialization</dfn> given a [=request=] (|request|), a string (|endpoint|) and an
+[=environment settings object=] (|settings|), run the following steps:
+
+1.  Let |blocked url| be |request|'s [=request/URL=].
+
+    Note: This is not |request|'s [=request/current URL=] in order to avoid redirect
+    information leak.
+
+2.  Set |blocked url|'s [=url/username=] to the empty string, and its [=url/password=] to `null`.
+
+3.  Set |serialized blocked url| be the result of executing the
+    [URL serializer](https://url.spec.whatwg.org/#concept-url-serializer) on |blocked url|
+    with the |exclude fragment flag| set.
+
+4.  Let |body| be a new object containing the following properties with keys:
+
+    * key: "`type`", "`worker initializatoin`".
+    * key: "`blocked-url`", value: |serialized blocked url|.
+
+5.  [Queue](https://w3c.github.io/reporting/#queue-report) |body| as "`coep`" for
+    |endpoint| on |settings|.
+
+To <dfn abstract-op>check a global object's embedder policy</dfn>, given a [=global object=]
+(|global|), a [=environment settings object=] (|owner|) and a [=request=] (|request|), then
+
+1.  If |global| is a {{SharedWorkerGlobalScope}} or |global| is a {{ServiceWorkerGlobalScope}}, then
```

Comment:
This is fine for this patch, but do we really want to punt entirely on `SharedWorker`/`ServiceWorker`? I wonder if we need to do something to ensure that they can't be used as bridges between COEP and non-COEP pages?

Reply:
I think the difference is that SharedWorker/ServiceWorker forms its own browser context group but DedicatedWorker doesn't.

Reply:
They form their own agent cluster and more importantly don't have a single owner.

Reply:
Browsing context groups are only for holding top-level browsing contexts and agent clusters whose "root agent" is a window agent.